p2p/peer.go

```diff
@@ -174,13 +191,73 @@ func newPeer(
 	return p
 }
 
+func (p *peer) streamReadLoop(streamID byte, stream transport.Stream) {
```

Comment:
I would be against, at this point, implementing parallel processing of messages. This is a complex issue, discussed in https://github.com/cometbft/cometbft/issues/3250.

But from the current design, it appears that we need this architecture...